### DISCUSSION TITLE
Dependent Microsoft.Bcl.AsyncInterfaces version 1.0.0 vs Microsoft.Bcl.AsyncInterfaces version 8.0.0

### DISCUSSION THREAD

**Discussion by ajbouffard** [hardkoded/puppeteer-sharp#2425] [Q&A]
My VS2022 64bit (17.6.5) multi-project C# solution built and ran fine until I added PuppeteerSharp to one of my projects. Now at runtime, when that project tries to load, I get the following runtime exception, even though my solution uses Microsoft.Bcl.AsyncInterfaces version 8.0.0 . How do I fix my solution to force the more recent version of this dll to be used? Is it just a question of editing my app.config with a new binding redirect?

Exception thrown: 'System.IO.FileNotFoundException' in System.Private.CoreLib.dll
An exception of type 'System.IO.FileNotFoundException' occurred in System.Private.CoreLib.dll but was not handled in user code
Could not load file or assembly 'Microsoft.Bcl.AsyncInterfaces, Version=1.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. The system cannot find the file specified.

**Comment by kblok** [comment]
That's a good question. I'm wondering whether we should create a .NET 8 specific bundle.
